platform/ATParser.cpp

```diff
@@ -0,0 +1,371 @@
+/* Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @section DESCRIPTION
+ *
+ * Parser for the AT command syntax
+ *
+ */
+
+#include "ATParser.h"
+#include "mbed_poll.h"
+#include "mbed_debug.h"
+
+#define _LF 10
```

Comment:
+1 for discouraging using prefix ``_``, I only pinpointed that some headers in our framework as you wrote might be unhelpful for such a short common macros

> If we're being paranoid about CR and LF already existing as macros, maybe just put #undef first?

Could be or using an unnamed namespace static const variable in this case ? :-)